modules/renter/contractor/allowance.go

```diff
@@ -147,12 +148,10 @@ func (c *Contractor) SetAllowance(a modules.Allowance) error {
 	c.mu.Lock()
 	c.allowance = a
 	c.contracts = newContracts
-	// update metrics
-	var spending types.Currency
-	for _, contract := range c.contracts {
-		spending = spending.Add(contract.RenterFunds())
+	// if the currentPeriod was previously unset, set it now
+	if c.currentPeriod == 0 {
+		c.currentPeriod = periodStart
```

Comment:
should it only be updating to a new period if there isn't currently a period? Should it update any time that the renewWindow changes?

hmm. I guess that's probably too much detail/stress for this PR. We'll leave it for now but I'm thinking that we'll be coming back to it in the future.